### DISCUSSION TITLE
Install 4.22 on Debian Bullseye

### DISCUSSION THREAD

**Discussion by molfine** [i3/i3#5362] [Q&A] [answered]
Hi all,

I am excited about the new release and only through it found that the official Bullseye packe is on version 4.19.1-1.

I tried adding the [custom repositories](https://i3wm.org/docs/repositories.html) provided by Michael, however now I am running into issues that the dependencies (specifically libc6) are too old. 

What is the best way of solving this? Looks like the required versions are in bookworm or above. What is the best way to get the latest i3 version?

Thanks very much!

Daniel

**Comment by okraits** [comment]
- You can use sur5r's Debian repository at https://debian.sur5r.net/i3/ (mentioned in https://i3wm.org/docs/repositories.html under `Ubuntu repository`), it contains packages for Debian stable as well. You just have to wait until he updates the packages.
- You can build i3 from source. For instructions, look here: https://github.com/Airblader/i3/wiki/Building-from-source

**Reply by molfine** [reply]
Thank you - did not realize I could use the sur5r repos as well.

**Comment by MarrekNozka** [comment, marked as answer]
My solution in like cases: 

If package is in Sid I download Source Package: https://packages.debian.org/source/sid/i3-wm .

This files:
```
i3-wm_4.22-1.dsc
i3-wm_4.22.orig.tar.xz
i3-wm_4.22.orig.tar.xz.asc
i3-wm_4.22-1.debian.tar.xz
```

```bash
mkdir i3build
cd i3build
wget http://deb.debian.org/debian/pool/main/i/i3-wm/i3-wm_4.22-1.debian.tar.xz http://deb.debian.org/debian/pool/main/i/i3-wm/i3-wm_4.22.orig.tar.xz.asc http://deb.debian.org/debian/pool/main/i/i3-wm/i3-wm_4.22-1.dsc http://deb.debian.org/debian/pool/main/i/i3-wm/i3-wm_4.22.orig.tar.xz
```




and make build debian package from source:

```bash

dpkg-source -x i3-wm_4.22-1.dsc
cd i3-wm-4.22
sudo apt build-dep i3
dpkg-buildpackage -b -uc -us
cd ..
sudo apt install ./*.deb
```

It work (for me :-)!

**Reply by molfine** [reply]
Thanks Marrek - this worked well. Appreciate the detail.
